### PR TITLE
Rewrite a deep nested ternary `makeAlign`

### PR DESCRIPTION
**Pull request by fisker**
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/document/doc-printer.js [hunk 1]
```diff
@@ -19,17 +19,27 @@ function makeIndent(ind, options) {
...
+
+  if (n.type === "root") {
+    return { ...ind, root: ind };
+  }
+
+  return generateInd(
+    ind,
+    { type: typeof n === "string" ? "stringAlign" : "numberAlign", n },
+    options
+  );
```
Comment (brodybits) [inline src/document/doc-printer.js:42]: I would probably favor compacting it something like this:

```js
function makeAlign(ind, n, options) {
  if (n === -Infinity) {
    return ind.root || rootIndent();
  } else if (n < 0) {
    return generateInd(ind, { type: "dedent" }, options);
  } else if (!n) {
    return ind;
  } else if (n.type === "root") {
    return { ...ind, root: ind };
  }
  // else
  return generateInd(
    ind,
    { type: typeof n === "string" ? "stringAlign" : "numberAlign", n },
    options
  );
}
```

Comment (fisker) [reply src/document/doc-printer.js:42]: Why use useless else?

Comment (fisker) [reply src/document/doc-printer.js:42]: The `else if` is the same as the last `else` you omitted.

Comment (fisker) [reply src/document/doc-printer.js:42]: Actually with this option https://eslint.org/docs/rules/no-else-return#options-74, your version will be reported.

Comment (brodybits) [reply src/document/doc-printer.js:42]: Understood. I was hoping to get this block more compact, somehow. How about this:

```js
function makeAlign(ind, n, options) {
  if (n === -Infinity) return ind.root || rootIndent();

  if (n < 0) return generateInd(ind, { type: "dedent" }, options);

  if (!n) return ind;

  if (n.type === "root") return { ...ind, root: ind };

  return generateInd(
    ind,
    { type: typeof n === "string" ? "stringAlign" : "numberAlign", n },
    options
  );
}
```

Comment (fisker) [reply src/document/doc-printer.js]: I never use this style.

**Pull request merged**